src/ui/components/ExplorerHome/ExplorerHome.js

```diff
@@ -88,77 +98,164 @@ const useStyles = makeStyles((theme) => ({
+    },
   });
-  const {participants, organizations, bots, projects} = checkboxes;
 
-  const handleChange = (event) => {
-    setCheckboxes({...checkboxes, [event.target.name]: event.target.checked});
+  let credTimelineSummary = [];
+  const credAndGrainSummary = {
+    totalCred: 0,
+    totalGrain: 0,
+    avgCred: 0,
+    avgGrain: 0,
   };
 
-  const data = [
-    {title: "Cred This Week", value: 610},
-    {title: "Grain Harvested", value: "6,765g"},
-    {title: "Active Participants", value: 13},
-    {title: "Grain per Cred", value: "22g"},
-  ];
+  // do the summaries of the current page of participants
+  for (const participant of tsParticipants.currentPage) {
+    // add this node's cred to the summary graph
+    credTimelineSummary = participant.credPerInterval.map(
+      (total, i) => (credTimelineSummary[i] || 0) + total
+    );
 
-  const createData = (username, cred, grain, chart) => ({
-    username,
-    cred,
-    grain,
-    chart,
-  });
+    credAndGrainSummary.totalCred += participant.cred;
+    credAndGrainSummary.totalGrain += Number(participant.grainEarned);
+  }
 
-  // sort by cred amount, highest to lowest
-  const nodes = sortBy(initialView.participants(), (n) => -n.cred);
-  // create an array of 0s for the cred summary graph at the top of the page
-  let credTimelineSummary = initialView.intervals().map(() => 0);
+  if (tsParticipants.currentPage.length > 0) {
+    credAndGrainSummary.avgCred =
+      credAndGrainSummary.totalCred / tsParticipants.currentPage.length;
+    credAndGrainSummary.avgGrain =
+      credAndGrainSummary.totalGrain / tsParticipants.currentPage.length;
+  } else {
+    credAndGrainSummary.avgCred = 0;
+    credAndGrainSummary.avgGrain = 0;
+  }
 
-  const rows = nodes.map((node) => {
-    const {credPerInterval} = node;
+  const summaryInfo = [
+    {title: "Cred This Week", value: 610, className: classes.credCircle},
+    {
+      title: `${currencyName}`,
+      value: `6,765${currencySuffix}`,
+      className: classes.grainCircle,
+    },
+    {
+      title: "Active Participants",
+      value: allParticipants.length,
+      className: classes.participantCircle,
+    },
+    {
+      title: `${currencyName} per Cred`,
+      value: `22${currencySuffix}`,
+      className: classes.grainPerCredCircle,
+    },
+  ];
 
-    // add this node's cred to the summary graph
-    credTimelineSummary = credTimelineSummary.map(
-      (total, i) => credPerInterval[i] + total
+  const filterIdentities = (event: SyntheticInputEvent<HTMLInputElement>) => {
+    // fuzzy match letters "in order, but not necessarily sequentially"
+    const filterString = event.target.value
+      .trim()
+      .toLowerCase()
+      .split("")
+      .join("+.*");
+    const regex = new RegExp(filterString);
+
+    tsParticipants.createOrUpdateFilterFn("filterIdentities", (participant) =>
+      regex.test(participant.identity.name.toLowerCase())
     );
+  };
+
+  const handleCheckboxFilter = (event) => {
```

Comment:
These should be wrapped in a `useCallback` I believe

Reply:
In this case, I don't think `useCallback` would buy us anything. It's more beneficial for eliminating needless rerenders of child components. Re-defining these functions every cycle is very low cost, and wrapping them in `useCallback` mostly, if not entirely, offsets the cost benefit.

If you're interested, here are some decent write-ups:
https://kentcdodds.com/blog/usememo-and-usecallback
https://blog.hackages.io/react-hooks-usecallback-and-usememo-8d5bb2b67231

Reply:
Thanks!